crates/remora/src/proxy/core.rs

```diff
@@ -238,28 +276,51 @@ mod tests {
             proxy_id, executor, mode, store, rx_proxy, tx_results, metrics,
         );
 
+        let transactions = if signed {
```

Comment:
shouldn't all transactions be signed at this stage?

Reply:
pre_execute is a helper function for the tests in this file. It generates some transactions, sends them to a proxy, waits for the output, and verifies that the output matches what we expect. (this was already the case before my changes). I've modified this function to allow us to test both the signed case (when we send correctly signed transactions to the proxy, in which case we expect them to be executed successfully) and the unsigned case (when we send unsigned transactions, in which case we expect them not to be executed). `signed` here is an argument that allows us to switch between the two cases.